### ISSUE TITLE
Useless `if-cond` should not trigger `useless-var` messaging

### ISSUE THREAD

**Issue by mduvall** [bbatsov/rubocop#1981]
The code below returns `W: Useless assignment to variable - foo.`. This is not so much a useless assignment as a useless condition (since both branches assign the variable to the same evaluated value).

```ruby
def foo
  if bar
    foo = 1
  else
    foo = 1
  end

  foo.bar.baz
end
```

**Comment by alexdowad** [comment]
I'm opening a PR which fixes this.
